lib/graphql/query/arguments.rb

```diff
@@ -10,8 +10,14 @@ class Arguments
       def initialize(values, argument_definitions:)
         @original_values = values
         @argument_values = values.inject({}) do |memo, (inner_key, inner_value)|
-          string_key = inner_key.to_s
-          arg_defn = argument_definitions[string_key]
+          arg_defn = argument_definitions[inner_key.to_s]
+
+          string_key = if arg_defn.as != nil
+            arg_defn.as.to_s
```

Comment:
good question. I'm tempted to leave at run time. I feel it might be confusing if someone is for example, comparing arguments together, or doing something like `if argument.as == :aSymbol`